clientv3/README.md

```diff
@@ -0,0 +1,64 @@
+# etcd/clientv3 [![Godoc](http://img.shields.io/badge/go-documentation-blue.svg?style=flat-square)](https://godoc.org/github.com/coreos/etcd/clientv3)
+
+`etcd/clientv3` is the official Go etcd client for v3.
+
+
+
+## Install
+
+```bash
+go get github.com/coreos/etcd/clientv3
+```
+
+
+## Accessing etcd
+
+Create client using `clientv3.New`:
+
+```go
+cli, err := clientv3.New(clientv3.Config{
+	Endpoints:   []string{"localhost:12378", "localhost:22378", "localhost:32378"},
+	DialTimeout: 5 * time.Second,
+})
+if err != nil {
+	log.Fatal(err)
+}
+defer cli.Close()
+```
+
+etcd v3 uses [`gRPC`](http://www.grpc.io) for remote procedure calls. And `clientv3` uses
+[`grpc-go`](https://github.com/grpc/grpc-go) to connect to etcd.
+
+1. Make sure to close the client after using it. `clientv3.Client.Close` method closes the client's `gRPC` connection. Otherwise it will leak goroutines.
+2. `grpc-go` has its own [retrying mechanism](https://github.com/grpc/grpc-go/blob/master/clientconn.go#L463). Having no `DialTimeout` in `Config` would block forever until it establishes successful `gRPC` connections.
+
+```go
+// without DialTimeout
+// Reconnecting to "localhost:12378"
+// Reconnecting to "localhost:12378"
+// Reconnecting to "localhost:22378"
+// [... until it succeeds]
+
+// with DialTimeout
+// Reconnecting to "localhost:132378"
+// grpc: timed out trying to connect
+```
+
+To specify client request timeout, pass `context.WithTimeout` to APIs:
+
+```go
+ctx, cancel := context.WithTimeout(context.Background(), timeout)
+resp, err := kvc.Put(ctx, "sample_key", "sample_value")
+select {
```

Comment:
this is not quite right since kvc.Put is a synchronous function call. User should check the error instead of trying to receive from ctx chan

```go
ctx, cancel := context.WithTimeout(context.Background(), timeout)
resp, err := kvc.Put(ctx, "sample_key", "sample_value")
cancel()
if err != nil {
    // handle error!
}
// use the response

```